client/src/components/Navigation.tsx

```diff
@@ -0,0 +1,312 @@
+                stroke="currentColor"
+                strokeWidth={strokeWidth}
+              >
+                <path
+                  strokeLinecap="round"
+                  strokeLinejoin="round"
+                  d={symbol}
+                />
+              </svg>
+            </div>
+
+            <span>{name}</span>
+          </div>
+        </a>
+      </li>
+    );
+  };
+
+  NavButton.propTypes = {
+    name: PropTypes.string,
+    url: PropTypes.string,
+    symbol: PropTypes.string,
+    strokeWidth: PropTypes.number,
+    viewBox: PropTypes.string,
+  };
+
+  const NavSymbols = ({navButtons}) => {
+    return (
+      <ul className="space-y-7 mt-30 ml-[rem] flex flex-col items-center mb-7">
+        {
+          navButtons.map((values, index) => (
+            <NavButton key = {index}
+                name={values.name}
+                url={values.url}
+                symbol={values.symbol}
+                strokeWidth={values.strokeWidth}
+                viewBox={values.viewBox}
+              />
+          ))
+        },
+      </ul>
+    );
+  };
+
+  NavSymbols.propTypes = {
+    navButtons: PropTypes.array,
+  };
+
+
+  return (
+    <div className="w-full h-full bg-gray-100">
+      <div className="flex flex-no-wrap">
+        <div className="absolute md:relative w-[14rem] h-screen  bg-zinc-900 hidden md:block z-10 overflow-auto">
+          <div className="h-12 w-full flex flex-col items-center">
+            <button
+              onClick={() => navigate('/')}
+              className="bg-transparent border-transparent justify-center items-center"
+            >
+              <img src={logo} className="mt-[5rem] w-[6rem]" alt="/" />
+            </button>
+          </div>
+          <NavSymbols navButtons={buttons}/>
+        </div>
+
+
+        <div
+          className={
+            showMobileMenu
+              ? 'w-full h-full absolute z-40  transform  translate-x-0 '
+              : 'w-full h-full absolute z-40  transform -translate-x-full'
+          }
+          id="mobile-nav"
+        >
+          <div
+            className="bg-gray-800 opacity-50 absolute h-full w-full md:hidden"
+            onClick={() => {
+              setShowMobileMenu(!showMobileMenu);
+            }}
+          />
+          <div className="absolute z-40 sm:relative w-[14rem] md:w-96 shadow bg-zinc-900 md:hidden transition duration-150 ease-in-out h-full">
+            <div className="flex flex-col justify-between h-full w-full overflow-auto">
+              <div>
+                <div className="flex items-center justify-between px-6">
+                  <div className="h-12  w-full flex flex-col ml-6 items-center">
+                    <button className="flex flex-col  bg-transparent border-transparent justify-center items-center">
+                      <img src={logo} className="mt-[5rem] w-[6rem]" alt="/"
+                        onClick={() => navigate('/')}/>
+                    </button>
```

Comment:
Usually when it comes to `onClick`s, they should almost always belong to `button` elements (with rare exceptions such as the `div` above to close the menu), as a button is something that is inherently clickable. So this `onClick` should be a property of the button.

If you're noticing that moving the onClick to the button lets the user click outside the img and still navigate to the homepage, then you could move the `mt-[5rem]` class to that button as well.

Reply:
I put the onClick on the image because it did not work on the button